apache/datadog_checks/apache/apache.py

```diff
@@ -29,6 +30,21 @@ class Apache(AgentCheck):
 
     RATES = {'Total kBytes': 'apache.net.bytes_per_s', 'Total Accesses': 'apache.net.request_per_s'}
 
+    SCOREBOARD_KEYS = {
```

Comment:
I see two possible approaches for reporting this data.
1. The one that you implemented where each "worker status" gets its own metric. The benefits are clear, though it's a bit hard if you want to get all metrics in an area graph.
2. The second approach would be to report a single metric (let's say `apache.scoreboard.workers_by_status`) that is tagged by its `worker_status`. The benefit of this approach is that it'll be much easier to make complicated queries like 
    a. Show an area graph with all the workers status.
    b. Show the total count of workers that are not disabled.
The drawback is that if you graph the plain metric `apache.scoreboard.workers_by_status` with the default `avg` space-aggregator, you'd get the average number of worker per status which doesn't make a lot of sense

The first option is perfectly fine in the general case but as you rightfully the mentioned the benefit of having a stacked-area graph, I believe the second option would work best.

Reply:
It was indeed an implementation detail I wasn't sure about. The reason I implemented 1 instead of 2 was because of other metrics I am familiar with that (to me at least) are similar and are implemented with each state/type as a separate metric. For example the CPU usage is split in 6 different metrics (`system.cpu.idle`, `system.cpu.user`, etc.). And the mysql queries (`mysql.performance.queries`) only reports the total queries and each query type isn't a tag but a separate metric (`mysql.performance.com_select`, `mysql.performance.com_insert`, etc.). 

The only example of metrics that I use and could think of while writing this are the haproxy metrics that tag each metric with a different `haproxy_service` to indicate which frontend/backend in haproxy the metric belongs to. In apache this would translate to different vhosts but on the `?auto` version of the mod status page you aren't able to know which vhost a worker is busy with. 

Thinking about the examples I knew of other integrations doing something similar it seemed that using different metrics for each state instead of tagging the states would be preferred. 

> b. Show the total count of workers that are not disabled.

Yeah excluding one or some states would be a lot easier with tagging, than it would be by combining metrics. But I feel like the discoverability of the different states is a lot harder, and as you stated the data you would get by not using the tags would be garbage by default. So when you were just trying each metric that the Apache integration provides by adding them to a timeseries widget in a dashboard (what I personally do to find useful metrics) it wouldn't be clear what its use is. 

> though it's a bit hard if you want to get all metrics in an area graph.

I feel like this could be easily fixed by adding an area timeseries with each state visible to the preset Apache overview dashboard. If a user doesn't know about the 'Advanced...' button to create stacked areas from different metrics, they could just copy the timeseries and paste it in their own dashboard. 

> The first option is perfectly fine in the general case but as you rightfully the mentioned the benefit of having a stacked-area graph, I believe the second option would work best.

I believe the best option would be the one that is the most consistent across other integrations. I am more aware of integrations that use the pattern from option 1 for metrics/values like this. But if 2 is used more consistently across integrations or preferred for newer metrics, I would change it to tags instead of separate metrics.

Reply:
We have exemple of both usage and I agree with you, option 1 is a bit more common as it's easier to make sense of it at the cost of a slight UX issue with stacked area graphs.

I'm perfectly fine with option 1, let's leave it like that 👍